CryptoTechReminderSystem/Gateway/HarvestGateway.cs

```diff
@@ -18,35 +19,49 @@ public HarvestGateway(string address, string token)
             _client = new HttpClient { BaseAddress = new Uri(address) };
             _token = token;
         }
+        
+        private async Task<JObject> GetApiResponse(string address)
+        {
+            var response = await _client.GetAsync(address);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+        }
 
-        public IList<Developer> Retrieve()
+        public IList<Developer> RetrieveDevelopers()
         {
             _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _token);
-            
-            var apiResponse = GetUsers().Result;
+ 
+            var apiResponse = GetApiResponse("/api/v2/users").Result;
             var users = apiResponse["users"];
-            IList<Developer> developers = new List<Developer>();
-            
-            foreach (var developer in users)
-            {
-                developers.Add(new Developer()
+            return users.Select(developer => new Developer
                 {
-                   Id = (int)developer["id"],
-                   FirstName = developer["first_name"].ToString(),
-                   LastName = developer["last_name"].ToString(),
-                   Email = developer["email"].ToString()
-                });
-            }
-            
-            return developers;
+                    Id = (int) developer["id"],
+                    FirstName = developer["first_name"].ToString(),
+                    LastName = developer["last_name"].ToString(),
+                    Email = developer["email"].ToString()
+                }
+            ).ToList(); 
         }
-        
-        private async Task<JObject> GetUsers()
+
+        public IEnumerable<TimeSheet> RetrieveTimeSheets()
         {
-            var requestUrl = "/api/v2/users";
-            var response = await _client.GetAsync(requestUrl);
-            var result = JObject.Parse(await response.Content.ReadAsStringAsync());
-            return result;
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _token);
+ 
+            var apiResponse = GetApiResponse("/api/v2/time_entries").Result;
+            var users = apiResponse["time_entries"];
+            return users.Select(timeEntry => new TimeSheet
```

Comment:
I think the `timeEntry` should have the same name as `TimeSheet`. There is inconsistency here.

Reply:
True, thanks for pointing out, fixed it! 849f554de57c8752e622a27d2ad79ded89d847c9